src/conventions/h-group/interpret-discard.js

```diff
@@ -32,6 +32,12 @@ export function interpret_discard(game, action, card) {
 
 	Basics.onDiscard(game, action);
 
+	if (playerIndex === state.ourPlayerIndex) {
+		// Possible self connections can be resolved.
+		for (const card of game.common.thoughts)
```

Comment:
Do we need to loop over all cards? I think only cards in our hand could have a self connection, but I might not be fully understanding.

Reply:
self_connection is set to true on other players' cards in other hands which could exist in our own finesse slot. So cards in our hand are the only ones we don't need to loop through.

Reply:
Okay, that makes sense. I think we can still reduce this to `for (const { order } of state.hands.flat())` (and then access `game.common.thoughts[order]`, for example, just so that we don't loop over cards that are no longer relevant.